pkg/conversion/converter.go

```diff
@@ -495,15 +495,6 @@ func (c *Converter) callCustom(sv, dv, custom reflect.Value, scope *scope) error
 // one is registered.
 func (c *Converter) convert(sv, dv reflect.Value, scope *scope) error {
 	dt, st := dv.Type(), sv.Type()
-	// Apply default values.
-	if fv, ok := c.defaultingFuncs[st]; ok {
```

Comment:
That might be OK for now, for expediency, but I don't think it's right for
the long term.

On Tue, Oct 27, 2015 at 10:06 AM, Yu-Ju Hong <notifications@github.com>
wrote:

> In pkg/conversion/converter.go
> <https://github.com/kubernetes/kubernetes/pull/16080#discussion_r43152368>
> :
>
> > @@ -495,15 +495,6 @@ func (c *Converter) callCustom(sv, dv, custom reflect.Value, scope *scope) error
> >  // one is registered.
> >  func (c *Converter) convert(sv, dv reflect.Value, scope *scope) error {
> >  	dt, st := dv.Type(), sv.Type()
> > -	// Apply default values.
> > -	if fv, ok := c.defaultingFuncs[st]; ok {
>
> Appropriate defaulting functions are called in individual (automatically
> generated or custom) conversion functions.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/16080/files#r43152368>.
>